### PR TITLE
feat: remove old project, start unit testing

### LINKED ISSUE #37: add unit tests

### PR DESCRIPTION
**Pull request by wisedev-pstach**

### CODE REVIEW

File: Raven.NET.Core.Tests/ProviderTests/RavenProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        var raven = _ravenWatcher.Object;
+        var name = "TestWatcher";
+        RavenCache.RavenWatcherCache.TryAdd(name, raven);
+        
+        //Act
+        var result = sut.GetRaven(name);
+        
+        //Assert
+        result.ShouldNotBeNull();
+        result.ShouldBeEquivalentTo(raven);
```
Comment (CebulMaciej) [inline Raven.NET.Core.Tests/ProviderTests/RavenProviderTests.cs]: ShouldBe is not enough? just a question :)

File: Raven.NET.Core.Tests/WatcherTests/RavenTypeWatcherTests.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+        _ravenSettingsProvider.Setup(x => x.GetRaven(watcherName)).Returns(settingsMock);
+        
+        //Act
+        sut.Create<TestSubjectEntity>(watcherName, nameof(TestSubjectEntity.Value),subject => true);
+        
+        //Assert
+        RavenCache.RavenTypeWatcherCache.ShouldContainKey(typeof(TestSubjectEntity));
+        var settings = (sut as RavenTypeWatcher)?._ravenSettings;
+        settings.LogLevel.ShouldBe(LogLevel.Trace);
```
Comment (CebulMaciej) [inline Raven.NET.Core.Tests/WatcherTests/RavenTypeWatcherTests.cs:78]: Github is hinting error, nice feature but here maybie this is true

**Review by CebulMaciej — APPROVED** [review]

File: Raven.NET.Core.Tests/ProviderTests/RavenProviderTests.cs (hunk 1 above)
Comment (wisedev-pstach) [reply Raven.NET.Core.Tests/ProviderTests/RavenProviderTests.cs]: Correct, I changed it ;)

File: Raven.NET.Core.Tests/WatcherTests/RavenTypeWatcherTests.cs (hunk 2 above)
Comment (wisedev-pstach) [reply Raven.NET.Core.Tests/WatcherTests/RavenTypeWatcherTests.cs:78]: This is mocked object, so its kind of impossible for this to be null, but I`ve added additional assertion (just to be 1212% sure :D ), this should remove github hint

**Pull request merged**